rollup.config.js

```diff
@@ -140,55 +140,99 @@ const makePlugins = (isProduction = false) => [
   isProduction ? terserMinified : terserPretty
 ].filter(Boolean);
 
-const config = {
-  input: './src/index.ts',
+const makeConfig = (input) => ({
+  input,
   external: externalTest,
   treeshake: {
     propertyReadSideEffects: false
   }
-};
+});
 
 export default [
   {
-    ...config,
-    plugins: makePlugins(false),
+    ...makeConfig('./src/index.ts'),
```

Comment:
We'll need to be able to use Rollup's code splitting feature. We'd use it by aliasing the core and main entries. It can then generate separate bundles that depend on one another.

This way people won't run into global state issues or duplicated code when they're using both, which is highly likely as we add more third party libraries